userspace/libscap/scap.c

```diff
@@ -1680,3 +1680,12 @@ uint64_t scap_get_driver_schema_version(scap_t* handle)
 {
 	return handle->m_schema_version;
 }
+
+scap_swap_endian scap_get_swap_endian(scap_t* handle)
+{
+	if(!handle->m_reader)
+	{
+		return SCAP_NATIVE_ENDIAN;
+	}
+	return handle->m_reader->m_swap_endian;
```

Comment:
I'm still getting a crash here, looks like `m_reader` is not initialised.
```
(gdb) p handle->m_reader
$3 = (scap_reader_t *) 0xc3890000c23be890
(gdb) p handle->m_reader->m_swap_endian
Cannot access memory at address 0xc3890000c23be8a0
(gdb) bt
#0  scap_get_swap_endian (handle=0x7ffff79d2047 <calloc+151>) at /libs/userspace/libscap/scap.c:1686
#1  0x000000000058afc2 in sinsp_evt::sinsp_evt (this=this@entry=0x7fffffffde50, inspector=inspector@entry=0x7fffffffdd70)
    at /libs/userspace/libsinsp/event.cpp:79
#2  0x00000000005e7ddb in sinsp::sinsp (this=this@entry=0x7fffffffdd70, static_container=static_container@entry=false, static_id="", static_name="",
    static_image="") at /libs/userspace/libsinsp/sinsp.cpp:73
#3  0x000000000057ef94 in main (argc=1, argv=0x7fffffffe6d8) at /libs/userspace/libsinsp/examples/test.cpp:70
```

Reply:
@Molter73 oh well... :sweat_smile: 
```
Program received signal SIGSEGV, Segmentation fault.
scap_get_swap_endian (handle=0x64657461657243) at /home/angelo/Projects/falcosecurity/libs/userspace/libscap/scap.c:1686
1686            if(!handle->m_reader)
(gdb) p handle
$2 = (scap_t *) 0x64657461657243
```